viewer/packages/sector-parser/src/DracoDecoderHelper.ts

```diff
@@ -0,0 +1,257 @@
+      },
+      decodedVertexBuffer
+    );
+
+    module._free(decodedBufferPointer);
+
+    return { decodedVertexBufferView: decodedVertexBuffer, vertexBufferDescriptor: vertexBufferDescriptor };
+  }
+
+  private computeVertexBufferDescriptor(
+    attributeDescriptors: AttributeDescriptor[],
+    dracoDataType: Map<number, DataType>,
+    numberOfVertices: number,
+    decoder: Decoder,
+    dracoMesh: Mesh,
+    decodedBufferPointer: number
+  ) {
+    let attributePointer = 0;
+    let attributeOffset = 0;
+    const vertexBufferDescriptor = attributeDescriptors.reduce<VertexBufferDescriptor>(
+      (accumulator, attributeDescriptor) => {
+        const { componentSize, numberOfComponents, dracoAttribute, gltfAccessor } = attributeDescriptor;
+        const dracoType = dracoDataType.get(gltfAccessor.componentType)!;
+        const byteLength = componentSize * numberOfComponents * numberOfVertices;
+        decoder.GetAttributeDataArrayForAllPoints(
+          dracoMesh,
+          dracoAttribute,
+          dracoType,
+          byteLength,
+          decodedBufferPointer + attributePointer
+        );
+
+        gltfAccessor.byteOffset = attributeOffset;
+
+        const attributeStride = componentSize * numberOfComponents;
+
+        attributePointer += byteLength;
+
+        accumulator[attributeDescriptor.attributeName] = { byteStride: attributeStride, byteOffset: attributeOffset };
+
+        attributeOffset += componentSize * numberOfComponents;
+        return accumulator;
+      },
+      {}
+    );
+    return vertexBufferDescriptor;
+  }
+
+  private getAttributeDescriptors(
+    json: GltfJson,
+    decoder: Decoder,
+    gltfAttributes: { [key: string]: number },
+    dracoAttributes: { [key: string]: number },
+    dracoMesh: Mesh
+  ): AttributeDescriptor[] {
+    return Object.keys(gltfAttributes).map(attributeName => {
+      const gltfAccessor = json.accessors[gltfAttributes[attributeName]];
+
+      const numberOfComponents = COLLECTION_TYPE_SIZES.get(gltfAccessor.type)!;
+      const componentSize = DATA_TYPE_BYTE_SIZES.get(gltfAccessor.componentType)!.BYTES_PER_ELEMENT;
+
+      const dracoAttribute = decoder.GetAttributeByUniqueId(dracoMesh, dracoAttributes[attributeName]);
+
+      return {
+        attributeName,
+        numberOfComponents,
+        componentSize,
+        gltfAccessor,
+        dracoAttribute
+      };
+    });
+  }
+
+  private decodeIndex(decoder: Decoder, mesh: Mesh, decoderModule: DecoderModule): Uint16Array | Uint32Array {
+    const numFaces = mesh.num_faces();
+    const numIndices = numFaces * 3;
+
+    let ptr: number;
+    let indices: Uint16Array | Uint32Array;
+
+    if (mesh.num_points() <= 2 ** 16) {
+      const byteLength = numIndices * Uint16Array.BYTES_PER_ELEMENT;
+      ptr = decoderModule._malloc(byteLength);
+      decoder.GetTrianglesUInt16Array(mesh, byteLength, ptr);
+      indices = new Uint16Array(decoderModule.HEAPU16.buffer, ptr, numIndices).slice();
```

Comment:
Yes, this needs to be done because the draco decoder module will have that memory freed up right after such that it can be reused for other sectors coming in.